### ISSUE TITLE
[question] On mpi4py

### ISSUE THREAD

**Issue by zyzhang1130** [hill-a/stable-baselines#944]
It is recommended in the documentation that if OpenMPI has issue with tensor flow then we should do pip uninstall mpi4py. Indeed this issue occurred for my case and I uninstalled mpi4py (the IDE (spyder4) kernel died and I have to close it and open again). But after that I could not import GAIL:

```python
from stable_baselines import GAIL

```
```bash
ImportError: cannot import name 'GAIL' from 'stable_baselines' (/home/user/anaconda3/lib/python3.7/site-packages/stable_baselines/__init__.py)

```

and when I do
```python
from stable_baselines.gail import ExpertDataset, generate_expert_traj

```
I got 

```bash
ModuleNotFoundError: No module named 'mpi4py'

```
So it seems I must have mpi4py installed. Is there any workaround (eg. some version of tensorflow works with mpi4py)? 

**System Info**
 * Installation method I used:
```bash
sudo apt-get update && sudo apt-get install cmake libopenmpi-dev python3-dev zlib1g-dev
pip install stable-baselines[mpi]

```
and I used 
```bash
pip uninstall mpi4py 

```

when mpi4py has issue with tensorflow

 * Python version:3.7.7
 * Tensorflow version:1.15.0

**Comment by araffin** [comment]
Hello,
you can consider using anaconda environments or docker to handle the installation.

The error traceback and os version are also missing from your question.

**Comment by zyzhang1130** [comment]
@araffin Hi, thanks for your reply. When mpi4py is installed the error messages are:
```bash
WARNING:tensorflow:
The TensorFlow contrib module will not be included in TensorFlow 2.0.
For more information, please see:
  * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
  * https://github.com/tensorflow/addons
  * https://github.com/tensorflow/io (for I/O related ops)
If you depend on functionality not listed there, please file an issue.

Creating environment from the given name, wrapped in a DummyVecEnv.
WARNING:tensorflow:From /home/user/Documents/Zeyu/stable-baselines/stable_baselines/common/tf_util.py:191: The name tf.ConfigProto is deprecated. Please use tf.compat.v1.ConfigProto instead.

WARNING:tensorflow:From /home/user/Documents/Zeyu/stable-baselines/stable_baselines/common/tf_util.py:200: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.

2020-07-14 23:08:53.982559: F tensorflow/stream_executor/lib/statusor.cc:34] Attempting to fetch value instead of handling error Internal: no supported devices found for platform CUDA
[jiafei-workstation:28259] *** Process received signal ***
[jiafei-workstation:28259] Signal: Aborted (6)
[jiafei-workstation:28259] Signal code:  (-6)
[jiafei-workstation:28259] [ 0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x12890)[0x7f01c077a890]
[jiafei-workstation:28259] [ 1] /lib/x86_64-linux-gnu/libc.so.6(gsignal+0xc7)[0x7f01c03b5e97]
[jiafei-workstation:28259] [ 2] /lib/x86_64-linux-gnu/libc.so.6(abort+0x141)[0x7f01c03b7801]
[jiafei-workstation:28259] [ 3] /home/user/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/_pywrap_tensorflow_internal.so(+0xadf98b4)[0x7f017ec018b4]
[jiafei-workstation:28259] [ 4] /home/user/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/_pywrap_tensorflow_internal.so(_ZN15stream_executor4port17internal_statusor6Helper5CrashERKN10tensorflow6StatusE+0x4e)[0x7f017eb05ade]
[jiafei-workstation:28259] [ 5] /home/user/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/_pywrap_tensorflow_internal.so(_ZNK10tensorflow9XlaDevice6clientEv+0xdd)[0x7f0179eaca8d]
[jiafei-workstation:28259] [ 6] /home/user/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/_pywrap_tensorflow_internal.so(_ZN10tensorflow9XlaDevice22GetDeviceContextLockedEv+0x43)[0x7f0179ead563]
[jiafei-workstation:28259] [ 7] /home/user/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/_pywrap_tensorflow_internal.so(_ZN10tensorflow9XlaDevice16UseGpuDeviceInfoEv+0x37)[0x7f0179eae2b7]
[jiafei-workstation:28259] [ 8] /home/user/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/_pywrap_tensorflow_internal.so(_ZN10tensorflow19XlaGpuDeviceFactory13CreateDevicesERKNS_14SessionOptionsERKSsPSt6vectorISt10unique_ptrINS_6DeviceESt14default_deleteIS8_EESaISB_EE+0x65c)[0x7f0179e9872c]
[jiafei-workstation:28259] [ 9] /home/user/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/../libtensorflow_framework.so.1(_ZN10tensorflow13DeviceFactory10AddDevicesERKNS_14SessionOptionsERKSsPSt6vectorISt10unique_ptrINS_6DeviceESt14default_deleteIS8_EESaISB_EE+0xed)[0x7f017306529d]
[jiafei-workstation:28259] [10] /home/user/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/_pywrap_tensorflow_internal.so(_ZN10tensorflow20DirectSessionFactory10NewSessionERKNS_14SessionOptionsEPPNS_7SessionE+0x32b)[0x7f017935102b]
[jiafei-workstation:28259] [11] /home/user/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/../libtensorflow_framework.so.1(_ZN10tensorflow10NewSessionERKNS_14SessionOptionsEPPNS_7SessionE+0xb0)[0x7f01730e6300]
[jiafei-workstation:28259] [12] /home/user/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/_pywrap_tensorflow_internal.so(TF_NewSession+0x24)[0x7f01767be204]
[jiafei-workstation:28259] [13] /home/user/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/_pywrap_tensorflow_internal.so(_ZN10tensorflow16TF_NewSessionRefEP8TF_GraphPK17TF_SessionOptionsP9TF_Status+0x12)[0x7f017611cc22]
[jiafei-workstation:28259] [14] /home/user/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/_pywrap_tensorflow_internal.so(+0x22a9af1)[0x7f01760b1af1]
[jiafei-workstation:28259] [15] python(_PyMethodDef_RawFastCallKeywords+0x1e0)[0x560b796cbc10]
[jiafei-workstation:28259] [16] python(_PyCFunction_FastCallKeywords+0x21)[0x560b796cbdb1]
[jiafei-workstation:28259] [17] python(_PyEval_EvalFrameDefault+0x4dee)[0x560b797375be]
[jiafei-workstation:28259] [18] python(_PyEval_EvalCodeWithName+0x2f9)[0x560b7967b2b9]
[jiafei-workstation:28259] [19] python(_PyFunction_FastCallKeywords+0x387)[0x560b796cb497]
[jiafei-workstation:28259] [20] python(_PyEval_EvalFrameDefault+0x14ea)[0x560b79733cba]
[jiafei-workstation:28259] [21] python(_PyEval_EvalCodeWithName+0x5da)[0x560b7967b59a]
[jiafei-workstation:28259] [22] python(_PyFunction_FastCallDict+0x400)[0x560b7967c610]
[jiafei-workstation:28259] [23] python(_PyObject_Call_Prepend+0x63)[0x560b7969ab93]
[jiafei-workstation:28259] [24] python(+0x16c0aa)[0x560b796d20aa]
[jiafei-workstation:28259] [25] python(_PyObject_FastCallKeywords+0x128)[0x560b796d2ca8]
[jiafei-workstation:28259] [26] python(_PyEval_EvalFrameDefault+0x55a8)[0x560b79737d78]
[jiafei-workstation:28259] [27] python(_PyEval_EvalCodeWithName+0x2f9)[0x560b7967b2b9]
[jiafei-workstation:28259] [28] python(_PyFunction_FastCallKeywords+0x387)[0x560b796cb497]
[jiafei-workstation:28259] [29] python(_PyEval_EvalFrameDefault+0x14ea)[0x560b79733cba]
[jiafei-workstation:28259] *** End of error message ***
Aborted (core dumped)


```

the os info are as follows:
* Operating System: Ubuntu 18.04.2 LTS
* Kernel: Linux 4.15.0-48-generic
* Architecture: x86-64

Also allow me to clarify when you say 'using anaconda environments' do you mean using conda install to install 'stable-baselines'? I didn't not find it available on anaconda cloud though.

**Comment by zyzhang1130** [comment]
@araffin Hi, I have identified the issue. Turns out it's my GPU out of memory. After I killed some processes it worked. Thanks a lot for replying my question anyways.
